event-streams/src/event-streams-v1.0-wd01.md

```diff
@@ -0,0 +1,710 @@
+property.
+
+If the requested partition does not exist or if the event log node denies the
+request, attaching the link MUST be rejected by the event log node.
+
+The event log node MAY bind the link a partition when none has been requested by
+the consumer.
+
+If the link becomes partition-bound, the event log node MUST set the
+[`event-streams-partition`](#461-event-streams-partition-link-property) link
+property to the bound partition.
+
+## 4.4 Partition annotations in consumer transfers
+
+The
+[`event-streams-source-partition`](#464-event-streams-source-partition-delivery-annotation)
+delivery annotation MUST be added to all transfers to consumers made over
+partition-agnostic links.
+
+The annotation is OPTIONAL for transfers over partition-bound links, and if
+present, its value MUST be equal to the `event-streams-partition` link property
+value. Otherwise, the transfer MUST be rejected.
+
+## 4.5 Consumer groups
+
+In scenarios where a group of multiple concurrent event consumers want to
+receive mutually exclusive subsets of events from a partitioned event log, it is
+desireable to restrict each partition to one (the "owning") receiver from the
+group, and for event consumers to shift ownership of the partition dynamically.
+Consumers might also want to use an external consensus mechanism to agree on
+which event processor owns which partition, rather than have the event log
+engine coordinate the assignments.  
+
+This specification does not prescribe any algorithm or strategy for reaching
+consensus on ownership of partitions amongst a group of consumers, but it
+provides mechanisms for realizing their outcomes.
+
+A consumer link MAY be bound to a consumer group by setting the
+[`event-streams-consumer-group`](#462-event-streams-consumer-group-link-property)
+link property to the name of the group during attach.
+
+The consumer group MAY be dynamically created during link attach or MAY require
+for there to be a such named pre-configured entity inside of event log node, in
+which case attaching the link MAY fail if such an entity does not exist.
+
+The event log node MUST NOT assign a consumer group if the consumer does not
+request such a binding. Negotiated links that are bound to consumer groups MUST
+also be bound to a partition; they cannot be partition-agile.
+
+The selection and binding of a partition to the consumer within a given consumer
+group follows the negotiation model explained earlier in this section.
+
+If consumer groups are being used, the event log node MUST allow at most one
+active 'receive' link for each combination of partition and consumer group.
+
+### 4.5.1 Event log node-assigned partition ownership
+
+If the event log node wants to force renegotiation of partition assignments
+amongst a consumer group, it needs to close all affected links and the
+respective consumers SHOULD attempt to establish a new link to request or
+receive new assignments.
```

Comment:
I think this part is still a bit confusing for me. 
* We can use only partition bound links which are always bound only to one particular partition
* When the event log node is assigning the partitions, the consumer in general does not know how many partitions will it have assigned and how many links does it need to open

So do we expect it to work in a way that the consumer tries to open new links _in a loop_ until the event-log-node runs out of partitions to assign?

Reply:
maybe the idea is about using the "Runtime information" which is not defined yet.
I read "Consumers and producers learn about an event log node's available partitions and their identifiers using a [runtimeinformation](#6-runtime-information) request." so we expect a way for a consumer to get information about available partitions before attaching links for consuming. At same time I can understand your point; thinking about Kafka, a consumer really doesn't know this information since the beginning. It just joins a group and partitions are assigned to it.

Reply:
Thinking about the `assign` in Kafka, you can specify partitions to be assigned. If the consensus algorithm logic between consumers is defined somewhere (so it's not provided with the consumer-group/subscribe logic), at some point they should agree on an assignment calling an `assign` with right partitions. Could be it the same in this case? After reaching a consensus, the consumers in the same group will attach more links specifying the right assigned partition for each of them?